src/richie/apps/courses/views.py

```diff
@@ -0,0 +1,32 @@
+from dal import autocomplete
+
+from cms.api import Page
+
+
+class PersonPageAdminAutocomplete(autocomplete.Select2QuerySetView):
+    """
+    Autocomplete view for Person search
+    """
+
+    def get_queryset(self):
+        """
+        Return queryset to use for returning autocomplete results.
+
+        Without any search keyword from ``self.q`` every results are returned, if
+        keyword is given results are filtered on it with insensitive ``contains``.
+        """
+        # Filter out results depending visitor
+        if not self.request.user.is_authenticated:
+            return Page.objects.none()
+
+        # Retrieve only draft Person pages
+        qs = Page.objects.filter(
+            publisher_is_draft=True,
+            person__isnull=False
+        )
+
+        # Perform autocompletion search on Person page
+        if self.q:
+            qs = qs.filter(title_set__title__icontains=self.q)
+
+        return qs.distinct().order_by("title_set__title")
```

Comment:
This will have weird behaviors on objects that have translations. :thinking: can you just add a comment if you don't see how to do it straightaway? don't waste time on it.

Reply:
There was weird behavior indeed. I fixed them with the right queryset method and args.